llm_bench/python/who_what_benchmark/whowhatbench/evaluator.py

```diff
@@ -145,8 +154,21 @@ def default_gen_answer(model, tokenizer, question, max_new_tokens, crop_question
 
         answers = []
 
-        for q in tqdm(questions.values, desc="Evaluate pipeline"):
-            answers.append(gen_answer_fn(model, self.tokenizer, q, self.max_new_tokens, self._crop_question))
+        if generation_config is None:
+            for q in tqdm(questions.values, desc="Evaluate pipeline"):
+                answers.append(gen_answer_fn(model, self.tokenizer, q, self.max_new_tokens, self._crop_question))
+        else:
+            with tqdm(total=len(questions.values)) as progress_bar:
+                batch = []
+                for q_idx, q in enumerate(questions.values):
+                    progress_bar.update(1)
+                    batch.append(q)
+                    if len(batch) == self.seqs_per_request or q_idx == len(questions.values) - 1:
+                        ans_batch = model.generate(batch, [generation_config] * len(batch))
+                        for ans in ans_batch:
+                            answers.append(ans.m_generation_ids[0])
```

Comment:
Why are token indices stored instead of text?

Reply:
This is, in fact, text in CB lib's API.